flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/TaskCheckpointingBehaviourTest.java

```diff
@@ -141,13 +141,18 @@ public void testBlockingNonInterruptibleCheckpoint() throws Exception {
 		assertNull(task.getFailureCause());
 	}
 
-	private void runTestDeclineOnCheckpointError(AbstractStateBackend backend) throws Exception{
-
+	private void runTestDeclineOnCheckpointError(AbstractStateBackend backend, boolean expectFailure) throws Exception {
```

Comment:
It feels as if the test logic is hidden in this method from the actual test case perspective.

Can we instead of expectFailure pass this::runTaskExpectFailure and this::runTaskExpectCheckpointDeclined? (probably BiConsumer?)

Or maybe inline this method altogether. I see the value rather limited. 

Last option is to use Roman's pattern in `ChannelStateWriterImplTest` 
```
    @Test
	public void testDeclineOnCheckpointErrorInAsyncPart() throws Exception {
       runWithResponder((task, checkpointResponder) -> runTaskExpectCheckpointDeclined(task, checkpointResponder));
	}
```

Reply:
I don't like the reliance on passing consumers everywhere, it makes code a spaghetti, when you have to traverse sometimes long call stacks to actually find out what your method is actually doing.

But indeed, inlining is better option to passing the parameter.